iModelCore/ecobjects/test/NonPublished/SchemaMergerTests.cpp

```diff
@@ -4493,6 +4493,284 @@ TEST_F(SchemaMergerTests, MergeRelationshipConstraints)
+      <ECSchema schemaName="Skimah" nameSpacePrefix="ski" version="01.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.2.0">
+            <ECClass typeName="Joint" isDomainClass="True">
+                <ECProperty propertyName="n" typeName="int"/>
+            </ECClass>
+            <ECClass typeName="Fastener" isDomainClass="True">
+                <ECProperty propertyName="n" typeName="int"/>
+            </ECClass>
+            <ECClass typeName="Weld" >
+                <ECProperty propertyName="p" typeName="int" displayLabel="p"/>
+            </ECClass>
+    
+            <ECRelationshipClass typeName="JointHasRandomThings" isDomainClass="True" strength="referencing" strengthDirection="forward">
+              <Source cardinality="(0,1)" polymorphic="True">
+                  <Class class="Joint" />
+              </Source>
+              <Target cardinality="(0,N)" polymorphic="True">
+                  <Class class="Weld"/>
+              </Target>
+          </ECRelationshipClass>
+      </ECSchema>)xml";
+  
+      Utf8CP schemaXml2 = R"xml(<?xml version="1.0" encoding="UTF-8"?>
+      <ECSchema schemaName="Skimah" nameSpacePrefix="ski" version="01.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.2.0">
+            <ECClass typeName="Joint" isDomainClass="True">
+                <ECProperty propertyName="n" typeName="int"/>
+            </ECClass>
+            <ECClass typeName="Fastener" isDomainClass="True">
+                <ECProperty propertyName="n" typeName="int"/>
+            </ECClass>
+            <ECClass typeName="Weld" >
+                <ECProperty propertyName="p" typeName="int" displayLabel="p"/>
+            </ECClass>
+    
+            <ECRelationshipClass typeName="JointHasRandomThings" isDomainClass="True" strength="referencing" strengthDirection="forward">
+              <Source cardinality="(0,1)" polymorphic="True">
+                  <Class class="Joint" />
+              </Source>
+              <Target cardinality="(0,N)" polymorphic="True">
+                  <Class class="Fastener"/>
+              </Target>
+          </ECRelationshipClass>
+      </ECSchema>)xml";
+
+    ECSchemaReadContextPtr leftContext = InitializeReadContextWithAllSchemas({schemaXml1}, nullptr, true);
+    bvector<ECN::ECSchemaCP> leftSchemas = leftContext->GetCache().GetSchemas();
+
+    ECSchemaReadContextPtr rightContext = InitializeReadContextWithAllSchemas({schemaXml2}, nullptr, true);
+    bvector<ECN::ECSchemaCP> rightSchemas = rightContext->GetCache().GetSchemas();
+
+    auto schemaMergeOptions = SchemaMergeOptions();
+    SchemaMergeResult result1;
+    EXPECT_EQ(BentleyStatus::ERROR, SchemaMerger::MergeSchemas(result1, leftSchemas, rightSchemas, schemaMergeOptions));
+    schemaMergeOptions.SetSkipValidation(true);
+    SchemaMergeResult result2;
+    EXPECT_EQ(BentleyStatus::SUCCESS, SchemaMerger::MergeSchemas(result2, leftSchemas, rightSchemas, schemaMergeOptions));

```

Comment:
Like you already mentioned, use an expected result schema to make sure this works as we expect